src/streamlink/plugins/zhanqi.py

```diff
@@ -62,25 +30,17 @@ class Zhanqitv(Plugin):
     def can_handle_url(self, url):
         return _url_re.match(url)
 
-    @classmethod
-    def stream_weight(cls, stream):
-        if stream in STREAM_WEIGHTS:
-            return STREAM_WEIGHTS[stream], "zhanqitv"
-        return Plugin.stream_weight(stream)
-
     def _get_streams(self):
         match = _url_re.match(self.url)
         channel = match.group("channel")
 
-        roomID = http.get(self.url, schema=_roomID_schema)
-
-        res = http.get(API_URL.format(roomID=roomID))
+        res = http.get(API_URL.format(channel))
         room = http.json(res, schema=_room_schema)
         if not room:
             self.logger.info("Not a valid room url.")
             return
 
-        if room["status"] != STATUS_ONLINE:
+        if room["status"] != 4:
```

Comment:
Why remove the constant? Seems to me that having the code read `room["status"] != STATUS_ONLINE` is more understandable than `room["status"] != 4`.